addons/website_sale/models/product_product.py

```diff
@@ -418,3 +418,69 @@ def _prepare_gmc_additional_info(self):
             gmc_info['item_group_id'] = self.product_tmpl_id.id
 
         return gmc_info
+    # ---------------------------------------------------------
+    # Product Ribbon Methods
+    # ---------------------------------------------------------
+
+    def _get_ribbon(self, product_prices):
+        """
+        Retrieve the appropriate ribbon for the product based on manual or automatic assignment.
+
+        This method first checks if a ribbon is manually set If no manual ribbon is set, it returns
+        appropriate ribbon based on the priority assigned in product.ribbon.
+
+        :param dict product_prices: A dictionary containing product pricing information.
+        :return: The ribbon that matches the product's criteria or an empty ribbon record.
+        :rtype: `product.ribbon` recordset.
```

Comment:
It's best to keep the return type consistent.
```suggestion
        :rtype: A sudoed `product.ribbon` recordset.
```